src/libs/ReportUtils.ts

```diff
@@ -216,278 +237,208 @@ function isCanceledTaskReport(report = {}, parentReportAction = {}) {
         return false;
     }
-    const report = allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] || {};
-    return report && report.ownerEmail === currentUserEmail;
+    const report = allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`];
+    return report?.ownerEmail === currentUserEmail;
 }
 
 /**
  * Whether the provided report is an Admin room
- * @param {Object} report
- * @param {String} report.chatType
- * @returns {Boolean}
  */
-function isAdminRoom(report) {
+function isAdminRoom(report: OnyxEntry<Report>): boolean {
     return getChatType(report) === CONST.REPORT.CHAT_TYPE.POLICY_ADMINS;
 }
 
 /**
  * Whether the provided report is an Admin-only posting room
- * @param {Object} report
- * @param {String} report.writeCapability
- * @returns {Boolean}
  */
-function isAdminsOnlyPostingRoom(report) {
-    return lodashGet(report, 'writeCapability', CONST.REPORT.WRITE_CAPABILITIES.ALL) === CONST.REPORT.WRITE_CAPABILITIES.ADMINS;
+function isAdminsOnlyPostingRoom(report: OnyxEntry<Report>): boolean {
+    return (report?.writeCapability ?? CONST.REPORT.WRITE_CAPABILITIES.ALL) === CONST.REPORT.WRITE_CAPABILITIES.ADMINS;
 }
 
 /**
  * Whether the provided report is a Announce room
- * @param {Object} report
- * @param {String} report.chatType
- * @returns {Boolean}
  */
-function isAnnounceRoom(report) {
+function isAnnounceRoom(report: OnyxEntry<Report>): boolean {
     return getChatType(report) === CONST.REPORT.CHAT_TYPE.POLICY_ANNOUNCE;
 }
 
 /**
  * Whether the provided report is a default room
- * @param {Object} report
- * @param {String} report.chatType
- * @returns {Boolean}
  */
-function isDefaultRoom(report) {
-    return [CONST.REPORT.CHAT_TYPE.POLICY_ADMINS, CONST.REPORT.CHAT_TYPE.POLICY_ANNOUNCE, CONST.REPORT.CHAT_TYPE.DOMAIN_ALL].indexOf(getChatType(report)) > -1;
+function isDefaultRoom(report: OnyxEntry<Report>): boolean {
+    return [CONST.REPORT.CHAT_TYPE.POLICY_ADMINS, CONST.REPORT.CHAT_TYPE.POLICY_ANNOUNCE, CONST.REPORT.CHAT_TYPE.DOMAIN_ALL].some((type) => type === getChatType(report));
 }
 
 /**
  * Whether the provided report is a Domain room
- * @param {Object} report
- * @param {String} report.chatType
- * @returns {Boolean}
  */
-function isDomainRoom(report) {
+function isDomainRoom(report: OnyxEntry<Report>): boolean {
     return getChatType(report) === CONST.REPORT.CHAT_TYPE.DOMAIN_ALL;
 }
 
 /**
  * Whether the provided report is a user created policy room
- * @param {Object} report
- * @param {String} report.chatType
- * @returns {Boolean}
  */
-function isUserCreatedPolicyRoom(report) {
+function isUserCreatedPolicyRoom(report: OnyxEntry<Report>): boolean {
     return getChatType(report) === CONST.REPORT.CHAT_TYPE.POLICY_ROOM;
 }
 
 /**
  * Whether the provided report is a Policy Expense chat.
- * @param {Object} report
- * @param {String} report.chatType
- * @returns {Boolean}
  */
-function isPolicyExpenseChat(report) {
+function isPolicyExpenseChat(report?: OnyxEntry<Report>): boolean {
```

Comment:
```suggestion
function isPolicyExpenseChat(report: OnyxEntry<Report>): boolean {
```

Reply:
I think this also need to be typed as `undefined` because for example 
<img width="714" alt="image" src="https://github.com/Expensify/App/assets/25230417/b3d73e77-7b87-40e6-8ab2-52cb7c4987c3">
<img width="742" alt="image" src="https://github.com/Expensify/App/assets/25230417/e4b3328e-66d0-485b-b159-d31a743e48bc">
and its because `report?.parentReportID` can be `undefined`

Reply:
Ah, okay. So, would it better to expect `report` passed always? Also, how should we handle `null` here?
```suggestion
function isPolicyExpenseChat(report: OnyxEntry<Report> | undefined): boolean {
```

Reply:
`OnyxEntry` type is making Report `nullable` 😄

Reply:
It is good then 🙂